script.js

```diff
@@ -1,4 +1,246 @@
+    // Increment currentIndex
+    index += 1;
+  }
+  // Return the shuffled deck
+  return cards;
+}
+
+// Function to Create NEW DECK
+
+function createNewDeck() {
+  var shuffleDeckDone = shuffleDeck(makeDeckDone);
+  return shuffleDeckDone;
+}
+
+//////// got blackjack?
+
+function gotBlackjack(handArray) {
+  var playerCardOne = handArray[0];
+  var playerCardTwo = handArray[1];
+
+  var gotBlackjack = false;
+
+  if (
+    (playerCardOne.name == "ace" && playerCardTwo.rank >= 10) ||
+    (playerCardOne.rank >= 10 && playerCardTwo.name == "ace")
+  ) {
+    gotBlackjack = true;
+  } else {
+    gotBlackjack = false;
+  }
+  return gotBlackjack;
+}
+
+function sumHandScore(handArray) {
+  var sum = 0;
+  var i = 0;
+
+  while (i < handArray.length) {
+    var currentCard = handArray[i];
+    if (
+      currentCard.name == "jack" ||
+      currentCard.name == "queen" ||
+      currentCard.name == "king"
+    ) {
+      sum = sum + 10;
+    } else {
+      sum = sum + currentCard.rank;
+    }
+    i += 1;
+  }
+  return sum;
+  ``;
+}
+
+// Game MODES
+var gameStart = "game start";
+var gameCardsIssued = "cards issued";
+var gameScore = "game score";
+var gameModeNow = gameStart;
+var playDeck = "";
+
+//////////////////////////////////////////
+
+function main(input) {
+  var instruction;
+
+  // SUBMIT
+  if (gameModeNow == gameStart) {
+    // create the game deck
+
+    playDeck = createNewDeck();
+    console.log(playDeck);
+
+    // deal 2 cards to player and dealer respectively
+
+    playerHand.push(playDeck.pop());
+    dealerHand.push(playDeck.pop());
+    playerHand.push(playDeck.pop());
+    dealerHand.push(playDeck.pop());
+
+    console.log("player hand >>");
+    console.log(playerHand);
+    console.log("dealer hand >>");
+    console.log(dealerHand);
+
+    gameModeNow = gameCardsIssued;
+
+    instruction = "Cards are dealt. Press SUBMIT to see cards";
+
+    return instruction;
+  }
+
+  if (gameModeNow == gameCardsIssued) {
+    //test for blackjack
+
+    // check for blackjack
+    var playerHasBlackJack = gotBlackjack(playerHand);
+    var dealerHasBlackJack = gotBlackjack(dealerHand);
+
+    if (playerHasBlackJack == true || dealerHasBlackJack == true) {
+      if (playerHasBlackJack == true && dealerHasBlackJack == true) {
+        instruction = "Both have achieved a DRAW";
+      } else if (playerHasBlackJack == true && dealerHasBlackJack == false) {
+        instruction = "Player Blackjack WINs";
+      } else {
+        instruction = "Dealer Blackjack WINs";
+      }
+      console.log(instruction);
+    } else {
+      instruction = " no one has blackjack ";
+
+      console.log(instruction);
+
+      var playerSumHandScore = sumHandScore(playerHand);
+      var dealerSumHandScore = sumHandScore(dealerHand);
+
+      console.log("Player Hand >>", playerSumHandScore);
+
+      console.log("Dealer Hand >>", dealerSumHandScore);
+
+      if (playerSumHandScore == dealerSumHandScore) {
```

Comment:
You could create a helper function here e.g. `checkWinner(playerHand, computerHand)`